src/v3/test/integration/authenticator-verification-email.test.tsx

```diff
@@ -49,13 +53,18 @@ describe('Email authenticator verification when email magic link = undefined', (
 
     it('renders the initial form with resend alert box', async () => {
       const {
-        container, findByText,
+        container, findByText, findByRole,
       } = await setup({
         mockResponse: authenticatorVerificationEmail,
+        widgetOptions: { features: { autoFocus: true } },
       });
 
       // renders the form
       await findByText(/Verify with your email/);
+      const codeEntryBtn = await findByRole(
+        'button', { name: 'Enter a code from the email instead' },
+      ) as HTMLButtonElement;
+      await waitFor(() => expect(codeEntryBtn).toHaveFocus());
```

Comment:
Is the `waitFor` necessary because the focus behavior happens very minimally later than the completion of widget render (due to the effect hook)?

Reply:
Yea, without the `waitFor` it fails (sometimes), there is a slight delay between when the element renders and when the focus happens.